File: leetcode-04/solutions/python3/0150.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []

        for token in tokens:
            if token == "+":
                b = stack.pop()
                a = stack.pop()
                stack.append(a + b)
            elif token == "-":
                b = stack.pop()
                a = stack.pop()
                stack.append(a - b)
            elif token == "*":
                b = stack.pop()
                a = stack.pop()
                stack.append(a * b)
            elif token == "/":
                b = stack.pop()
                a = stack.pop()
                stack.append(int(float(a) / b))
            else:
                stack.append(int(token))

        return stack.pop()

File: leetcode-04/solutions/cpp/0150.cpp
class Solution {
 public:
  int evalRPN(vector<string>& tokens) {
    stack<int> stack;
    int a;
    int b;

    for (string& token : tokens) {
      if (token == "+") {
        helper(stack, a, b);
        stack.push(a + b);
      } else if (token == "-") {
        helper(stack, a, b);
        stack.push(a - b);
      } else if (token == "*") {
        helper(stack, a, b);
        stack.push(a * b);
      } else if (token == "/") {
        helper(stack, a, b);
        stack.push(a / b);
      } else {
        stack.push(stoi(token));
      }
    }

    return stack.top();
  }

 private:
  void helper(stack<int>& stack, int& a, int& b) {
    b = stack.top();
    stack.pop();
    a = stack.top();
    stack.pop();
  }
};